python/lsst/ctrl/bps/bps_core.py

```diff
@@ -300,78 +277,80 @@ def _create_science_graph(self):
+                # style='"filled,bold"',
+                style="filled",
+                label=".".join(task_def.taskName.split(".")[-2:]),
+            )
+            quantum = node.quantum
+
+            # Make dataset ref nodes for inputs
+            for ds_refs in quantum.inputs.values():
+                for ds_ref in ds_refs:
+                    ds_name = f"{ds_ref.datasetType.name}+{ds_ref.dataId}"
+                    if ds_name not in dsname_to_node_id:
+                        dcnt += 1
+                        fnode_name = f"ds{dcnt:06}"
+                        dsname_to_node_id[ds_name] = fnode_name
+                        fnode_label = pretty_dataset_label(ds_name)
                         self.sci_graph.add_node(
-                            fnode_name, node_type=FILENODE, label=fnode_desk, shape="box", style="rounded"
+                            fnode_name, node_type=FILENODE, label=fnode_label, shape="box", style="rounded"
                         )
-                        self.sci_graph.add_edge(fnode_name, tnode_name)
-                # Make nodes for outputs
-                for ds_refs in quantum.outputs.values():
-                    for ds_ref in ds_refs:
-                        ds_name = "%s+%s" % (ds_ref.datasetType.name, ds_ref.dataId)
-                        if ds_name not in dsname_to_node_id:
-                            ncnt += 1
-                            dcnt += 1
-                            dsname_to_node_id[ds_name] = ncnt
-                        fnode_name = "%06d" % dsname_to_node_id[ds_name]
-                        fnode_desk = pretty_dataset_label(ds_name)
+                    fnode_name = dsname_to_node_id[ds_name]
+                    self.sci_graph.add_edge(fnode_name, tnode_name)
+
+            # Make dataset ref nodes for outputs
+            for ds_refs in quantum.outputs.values():
+                for ds_ref in ds_refs:
+                    ds_name = f"{ds_ref.datasetType.name}+{ds_ref.dataId}"
+                    if ds_name not in dsname_to_node_id:
+                        dcnt += 1
+                        fnode_name = f"ds{dcnt:06}"
+                        dsname_to_node_id[ds_name] = fnode_name
+                        fnode_label = pretty_dataset_label(ds_name)
                         self.sci_graph.add_node(
-                            fnode_name, node_type=FILENODE, label=fnode_desk, shape="box", style="rounded"
+                            fnode_name, node_type=FILENODE, label=fnode_label, shape="box", style="rounded"
                         )
-                        self.sci_graph.add_edge(tnode_name, fnode_name)
+                    fnode_name = dsname_to_node_id[ds_name]
+                    self.sci_graph.add_edge(tnode_name, fnode_name)
 
         if "pipeline" in self.config:
             self.pipeline = self.config["pipeline"].split(",")
         else:
-            self.pipeline = pipeline
+            self.pipeline = pipeline_task_labels.keys()
```

Comment:
Thanks for pointing out the iterTaskGraph method.  Will change the code.  This may actually be going away or done a completely different way later as one of the reasons for it was to make the individual --init-only jobs with the right dependencies among them.  That code is being replaced by a single --init-only job in another ticket.  If the concept stays around after that ticket, then a different name will be chosen.  The history behind the name was that before pipeline yaml existed there was BPS yaml for the user to define a pipeline.  The user provided the pipeline labels in order to find other information in the BPS yaml to use to build the pipeline/QuantumGraph.